platform/solr/solr-factory/src/test/groovy/org/codice/solr/factory/impl/SolrHttpRequestRetryHandlerSpec.groovy

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.solr.factory.impl
+
+import org.apache.http.protocol.HttpContext
+import org.codice.spock.extension.ClearInterruptions
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import javax.net.ssl.SSLException
+
+class SolrHttpRequestRetryHandlerSpec extends Specification {
+  static final String CORE = "test_core"
+
+  static final long TIME_FACTOR = 1L
+  static final int MAX_RETRY_COUNT = 4;
+
+  def httpContext = Stub(HttpContext)
+
+  def 'test constructor with null core'() {
+    when:
+      new SolrHttpRequestRetryHandler(null)
+
+    then:
+      def e = thrown(IllegalArgumentException)
+
+      e.message.contains('invalid null core')
```

Comment:
❓Do we really need to verify the exception message (same comment as above)?

Reply:
Same reply as above